tokio/src/sync/tests/loom_semaphore_batch.rs

```diff
@@ -0,0 +1,193 @@
+use crate::sync::batch_semaphore::*;
+
+use futures::future::poll_fn;
+use loom::future::block_on;
+use loom::thread;
+use std::future::Future;
+use std::pin::Pin;
+use loom::sync::atomic::AtomicUsize;
+use std::sync::atomic::Ordering::SeqCst;
+use std::sync::Arc;
+use std::task::Poll::Ready;
+use std::task::{Context, Poll};
+
+#[test]
+fn basic_usage() {
+    const NUM: usize = 2;
+
+    struct Shared {
+        semaphore: Semaphore,
+        active: AtomicUsize,
+    }
+
+    async fn actor(shared: Arc<Shared>) {
+        let mut permit = Permit::new();
+        permit.acquire(1, &shared.semaphore).await.unwrap();
+        println!("acquired!");
+        let actual = shared.active.fetch_add(1, SeqCst);
+        assert!(actual <= NUM - 1);
+
+        let actual = shared.active.fetch_sub(1, SeqCst);
+        assert!(actual <= NUM);
+
+        permit.release(1, &shared.semaphore);
+    }
+
+    loom::model(|| {
+
+        println!("\n ------ iter ------ \n");
```

Comment:
```suggestion
```

Reply:
If you want output, it probably should be gated on `LOOM_LOG=1`.

Reply:
whoops, yeah, i meant to remove that but forgot. thanks.